students/adam_strong/lesson08/circle.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+"""
+The ultimate circle.. and sphere
+Circle(object) or Sphere(Circle), __init__(self,radius)
+"""
+
+import math
+
+class Circle(object):
+
+    def __init__(self, radius):
+        self.radius = radius
+
+    @classmethod
+    def from_diameter(cls, _diameter):
+        radius = _diameter/2
+        return cls(radius)
+
+    @property
+    def radius(self):
+        return self._radius
+    @radius.setter
+    def radius(self, val):
+        if val < 0:
+            raise ValueError('You cannot have a negative radius')
+        else:
+            self._radius = val
```

Comment:
there is no need for both `self.radius` and `self._radius` since you're using `radius` as a property.

you should also make sure to define `self._radius` in the init method.